armi/physics/fuelCycle/fuelHandlers.py

```diff
@@ -997,14 +1104,133 @@ def readMoves(fname):
                 newLoc = m.group(2)
                 enrichList = [float(i) for i in m.group(3).split()]
                 # old loading style, just assume that there is a booster as our surrogate
-                moves[cycle].append((oldLoc, newLoc, enrichList, None))
+                moves[cycle].append(AssemblyMove(oldLoc, newLoc, enrichList))
                 numMoves += 1
 
         f.close()
 
         runLog.info("Read {0} moves over {1} cycles".format(numMoves, len(moves.keys())))
         return moves
 
+    @staticmethod
+    def readMovesYaml(fname):
+        r"""
+        Read a shuffle file in YAML format.
+
+        Parameters
+        ----------
+        fname : str
+            Path to the YAML-formatted shuffle file.
+
+        Returns
+        -------
+        moves : dict
+            Mapping of cycle numbers to lists of
+            :class:`~armi.physics.fuelCycle.fuelHandlers.AssemblyMove` objects that
+            describe the shuffle sequence.
+        misloadSwaps : dict
+            Mapping of cycle numbers to lists of location-pair tuples describing
+            assemblies to be swapped in order to simulate a misload.
+        """
+        try:
+            with open(fname, "r") as stream:
+                yaml = YAML(typ="safe")
+                data = yaml.load(stream)
+        except DuplicateKeyError as e:
+            raise InputError(str(e)) from e
+        except OSError as ee:
+            raise RuntimeError(
+                f"Could not find/open repeat shuffle file {fname!r} in working directory {os.getcwd()}: {ee}"
+            ) from ee
+
+        if "sequence" not in data:
+            raise InputError("Shuffle YAML missing required 'sequence' mapping")
+
+        moves = {}
+        misloadSwaps = defaultdict(list)
+        # cycles may be provided in any order; verify only that there are no gaps
+        cycleNums = {int(c) for c in data["sequence"].keys()}
+        if cycleNums:
+            expected = set(range(min(cycleNums), max(cycleNums) + 1))
+            missing = sorted(expected - cycleNums)
+            if missing:
+                if len(missing) == 1:
+                    raise InputError(f"Missing cycle {missing[0]} in shuffle sequence")
+                raise InputError(f"Missing cycles {missing} in shuffle sequence")
+
+        for cycleKey, actions in data["sequence"].items():
```

Comment:
```suggestion
        # parse YAML file into shuffle data
        for cycleKey, actions in data["sequence"].items():
```

Reply:
The idea here with these three little suggestions is this method is just too long. We could break it up into smaller, helper methods. But I know you are on a deadline. I think the quicker solution is to add some high-level bullet-point comments.

I'm willing to hear other ideas, of course.

Reply:
added the comments! 

[ceec293](https://github.com/terrapower/armi/pull/2219/commits/ceec29390618d08d972964c41880182f276673cb)